### PR TITLE
Set Retention Policy for Influx and downsample data

### LINKED ISSUE #1661: Influx: Downsample with continuous query
Create retention rules and continuous query to down sample data
Determine amount of space needed for retention

### PR DESCRIPTION
**Pull request by utopiabound**
Retain influx data for 90 days
Downsample iml_stats to 30m average, keep raw (per-second) data for 1 day.
This should keep total data size fairly small.

Fixes #1661 

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/1668)

### CODE REVIEW

File: chroma_core/lib/service_config.py [hunk 1]
```diff
@@ -317,7 +317,43 @@ def _setup_influxdb(self):
         log.info("Creating InfluxDB database...")
         self.try_shell(["influx", "-execute", "CREATE DATABASE {}".format(settings.INFLUXDB_IML_DB)])
         self.try_shell(["influx", "-execute", "CREATE DATABASE {}".format(settings.INFLUXDB_STRATAGEM_SCAN_DB)])
+        self.try_shell(
```
Comment (ip1981) [inline chroma_core/lib/service_config.py]: I wonder if this needs to be updated for docker too.

Comment (jgrund) [inline chroma_core/lib/service_config.py]: Given we only store the most recent scan result for stratagem, I wonder if we need to have a retention policy for it.

File: chroma_core/lib/service_config.py [hunk 2]
```diff
@@ -317,7 +317,43 @@ def _setup_influxdb(self):
...
+                'CREATE RETENTION POLICY "a_quarter" ON "{}" DURATION 90d REPLICATION 1 SHARD DURATION 5d'.format(
+                    settings.INFLUXDB_IML_STATS_DB
+                ),
+            ]
+        )
+        self.try_shell(
+            [
+                "influx",
+                "-execute",
+                'CREATE CONTINUOUS QUERY "downsample" ON "{}" BEGIN SELECT mean(*) INTO "{}"."a_quarter".:MEASUREMENT FROM /.*/ GROUP BY time(30m) END'.format(
```
Comment (jgrund) [inline chroma_core/lib/service_config.py]: Is `mean` the correct aggregation function for every stat being stored?

**Comment by jgrund** [comment]
Since we are now creating a new measurement `iml_stats.a_quarter`, we will need to dynamically switch  between the default measurement and this measurement when the date range in grafana is > 1 day.

Will this cause an issue in grafana, i.e. is there a way we can do this easily with our current charts?

File: chroma_core/lib/service_config.py (hunk 2 above)
Comment (utopiabound) [reply chroma_core/lib/service_config.py]: Still need to add mean of differences for several data sets

**Comment by utopiabound** [comment]
screen

File: chroma_core/lib/service_config.py (hunk 1 above)
Comment (johnsonw) [reply chroma_core/lib/service_config.py]: What would be the downside of using the 90 day retention policy?

Comment (jgrund) [reply chroma_core/lib/service_config.py]: Stats get dropped after 90 days. Though they would probably be stale by that point anyway.

**Pull request merged**